packages/server/modules/workspaces/services/workspaceLimits.ts

```diff
@@ -0,0 +1,31 @@
+import { getPaginatedProjectModelsTotalCountFactory } from '@/modules/core/repositories/branches'
+import { getProjectDbClient } from '@/modules/multiregion/utils/dbSelector'
+import {
+  GetWorkspaceModelCount,
+  QueryAllWorkspaceProjects
+} from '@/modules/workspaces/domain/operations'
+
+// TODO: What is correct DI for a service that has to dynamically select a regional database?
+export const getWorkspaceModelCountFactory =
```

Comment:
This already existed before, but this doesn't look very optimized. Instead of calculating the totalCount separately for each project, there could have just been a single DB-wide "count all workspace models for this region" SQL query.

Reply:
agreed, I was a bit stumped by the "dynamic region" bit. Left a TODO, we should revisit this soon